Makefile

```diff
@@ -1,28 +1,46 @@
+####################
+# Common constants #
+####################
+
 TEST_PATH=testing
 TEST_REQUIREMENTS=test_requirements.txt
 NOTEBOOK_PATH=notebooks
 
+
+##################################
+# Virtual environment management #
+##################################
+
 install:
 	pip install -r $(TEST_REQUIREMENTS)
 
+freeze_requirements:
+	pip freeze > $(TEST_REQUIREMENTS)
 
-test:
-	nosetests $(TEST_PATH)
 
+#############################
+# Jupyter notebook commands #
+#############################
 
 jupyter_server:
 	jupyter notebook $(NOTEBOOK_PATH)
 
 sync_notebooks:
 	jupytext --sync $(NOTEBOOK_PATH)/*.ipynb
 
+pair_notebooks:
+	jupytext --set-formats ipynb,py:percent $(NOTEBOOK_PATH)/*.ipynb
+
 check_notebooks_synced:
 	jupytext --test-strict -x $(NOTEBOOK_PATH)/*.ipynb --to py:percent
 
 
-freeze_requirements:
-	pip freeze > $(TEST_REQUIREMENTS)
+#############################################
+# Commands for making sure code doesn't rot #
+#############################################
 
+test:
+	nosetests $(TEST_PATH)
```

Comment:
not really this PR but should we switch to pytest?

Reply:
I've certainly used pytest more than nosetests, so would feel more comfortable with it. What do you think @sumonahmedUoM ?

Reply:
Yes, we should switch to pytest, I am also comparatively more familiar with it. @elvijs @alexisboukouvalas